### PR TITLE
Correct directory path for caching nox installed dependencies in CI

### PR DESCRIPTION
**Pull request by pankajkoti**
# Description
## What is the current behavior?

The cache is not referring to the right directory path for caching nox installed environments.

closes: #1135 

## What is the new behavior?

Correct the path for nox dependencies to cache.

## Does this introduce a breaking change?
No

### CODE REVIEW

**Comment by pankajkoti** [comment]
Although the cache is right referring to the nox directory path now, the tests themselves take a lot longer as we can see in the pytest summary. We will need to check if we can optimise the tests run duration.

File: sql-cli/sql_cli/constants.py [hunk 1]
```diff
@@ -6,3 +6,5 @@
 DEFAULT_DAGS_FOLDER = ".airflow/dags"
 
 GENERATED_WORKFLOW_INCLUDE_DIRECTORY = "include"
+
+TEST_CONSTANT = 1
```
Comment (pankajkoti) [inline sql-cli/sql_cli/constants.py:10]:
```suggestion

```

**Comment by pankajkoti** [comment]
We can see an improvement in the Type-Check job where it is 3x faster now because of caching nox dependencies correctly.

Before: 2min41s
Screenshot 2022-11-23 at 9 30 02 PM

Now: 46s
Screenshot 2022-11-23 at 9 30 21 PM

**Comment by pankajkoti** [comment]
Attaching screenshots for showing cache size before and after this change and that nox does not reinstall the packages upon using the cache.

Cache size before: ~25MB
Screenshot 2022-11-23 at 7 39 45 PM

Cache size after: ~228MB
Screenshot 2022-11-23 at 7 38 55 PM

**Review by tatiana — APPROVED** [review]
Great improvement, @pankajkoti !

File: .github/workflows/ci-sql-cli.yaml [hunk 2]
```diff
@@ -44,7 +44,7 @@ jobs:
         with:
            path: |
              ~/.cache/pip
-             .nox
+             sql-cli/.nox
```
Comment (pankajkoti) [inline .github/workflows/ci-sql-cli.yaml:47]:
```suggestion
             sql-cli/.nox
             .local
```

File: .github/workflows/ci-sql-cli.yaml [hunk 3]
```diff
@@ -87,7 +87,7 @@ jobs:
         with:
           path: |
             ~/.cache/pip
-            .nox
+            sql-cli/.nox
```
Comment (pankajkoti) [inline .github/workflows/ci-sql-cli.yaml:90]:
```suggestion
            sql-cli/.nox
            .local/
```

**Review by kaxil — APPROVED** [review]

**Pull request merged**